client/src/components/SidebarButton/SidebarButton.tsx

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+
+import "intersection-observer";
+
+import { FloatingButton } from "src/components/FloatingButton/FloatingButton";
+
+interface SidebarButtonProps {
+  open_sidebar: () => void;
+  left: boolean;
+  focus: () => void;
+}
+
+export class SidebarButton extends React.Component<SidebarButtonProps> {
```

Comment:
Doesn't seem useful as-is. If the caller has to provide a `focus` call back anyway then it might as well just handle focusing from the `open_sidebar` callback. Outside of that it's just `FloatingButton` with some renaming and enforced defaults.

I suggested a component for this, but what I had in mind was for it to be responsible for all "sidebar" behaviour. Interface for the simple, stateless, version could look something like this
```
const Sidebar = ({is_open, close_callback, children}) => /* 
  Sidebar is responsible for:
    - standard styling of the sidebar
    - a close button
    - hiding it when not open
    - moving focus in when opening (returning focus, if necessary, is the parent's job via the `close_callback`). 
    
  The parent is responsible for:
    - holding the open state
    - some arbitrary control to open the sidebar
    - supplying arbitrary children to render inside the sidebar
    - potentially handling focus return on close
 */
```

A separate more special case `SidebarButton` component could extend that stateless version, taking on the responsibility for everything but the the actual child content.

Reply:
Out of scope for now though, can drop this component in this PR and save that work for whenever we need a separate sidebar (... so probably right away, just in the branch for the table of contents).